search/node-search_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package search
+
+import (
+	"errors"
+	"strconv"
+	"testing"
+
+	"gopkg.in/yaml.v3"
+)
+
+func performSearch(t *testing.T, filePath string, seeking Seeking, yamlPaths ...[]string) []*yaml.Node {
+	var foundNodes []*yaml.Node
+	baseNode, parseNode := MakeNode(filePath)
+	if parseNode != nil {
+		t.Fatalf(parseNode.Error())
+	}
+	for _, yamlPath := range yamlPaths {
+		node, searchError := findComponent(baseNode.Content[0], seeking, yamlPath...)
```

Comment:
Can we test through the public method FindKey?

Reply:
now using findkey